python/src/space/core/storage.py

```diff
@@ -473,7 +540,7 @@ def __init__(self, storage: Storage):
     self._txn_id = uuid_()
     # The storage snapshot ID when the transaction starts.
     self._snapshot_id: Optional[int] = None
-
+    self._branch = storage.current_branch
```

Comment:
We may not need self._branch here if we push things into storage's method, please see the following comments.

Reply:
see comment above about concurrent write to different branch